uiautomationutilities/perf_tool/readme.md

```diff
@@ -1,155 +1,164 @@
+- On local windows machine create directories to keep your base and target codemarker files. e.g. `C:\testdata\basefiles` and `C:\testdata\targetfiles`
+- Download base files from any of the build run by going to the artifacts, clicking on Perf and then clicking on basefiles. For example go to "https://dev.azure.com/IdentityDivision/IDDP/_build/results?buildId=705958&view=results", click on artifact button "2 published; 1 consumed", click on Perf and then download basefiles. 
+- Extract and put the raw basefiles in your local base file folder of your machine.
+- On local windows machine go to your target codemarker files location and download the codemarker files named as "PerfData*.txt" from the android device from location "/sdcard" : (Note: make sure that the test case has completed before running the download command)
+- Open the Perf C# tool in visual studio by opening the solution file `PerfIdentity.sln`
+- Build the project by clicking on Build -> Build Solution
+- Copy `PerfDataConfiguration.xml` into the build folder e.g. `bin\Release\netcoreapp3.1`
+- Run the application. Arguments definition can be seen in `command.cs` file.
+
+
+### Where is the raw Perfdata.txt stored in local run as well as in pipeline run
+- In local run, the Perfdata txt files can be stored on any directory which should be given to the C# tool as an argument. e.g. `C:\testdata\basefiles` and `C:\testdata\targetfiles`
 - On pipeline, we can find the txt files from artifact "Perf -> basefiles" or "Perf -> targetfiles"
 
 
-#### Where to find the intermediate files or output files
+### Where to find the intermediate files or output files
 - If output directory is default i.e. "." then the output files will be created in the directory where the applicaiton is running.
-- Default directory of output files: "bin\Release\netcoreapp3.1"
+- Default directory of output files: `bin\Release\netcoreapp3.1`
 
 
 ### Arguments to the tool:
-- 1. Directory where PerfData base files are present. Example value: "C:\testdata\basefiles"
-- 2. Directory where PerfData target files are present. Example value: "C:\testdata\targetfiles"
-- 3. Build id of the base task.
-- 4. Build id which should be written in the final Email report and used for going to artifact url. Example value: "1234"
-- 5. Device model to be written in the final Email report. Example value: "Pixel2"
-- 6. Device OS to be written in the final Email report. Example value: "API28"
-- 7. App name to be written in the Email report. Example value: "MSALTestApp"
-- 8. Email ID of the sender's account. Example value: "idlab1@msidlab4.onmicrosoft.com"
-- 9. Password of the sender's account.
--10. Email To list separated by comma
+- Directory where PerfData base files are present. Example value: `C:\testdata\basefiles`
+- Directory where PerfData target files are present. Example value: `C:\testdata\targetfiles`
```

Comment:
Does this run on non-windows machines?

Reply:
With some modifications it could but it has only been tested on Windows - https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/paul/update-perf-testing-documentation/uiautomationutilities/perf_tool/readme.md#steps-to-deploy-latest-version-of-c-tool-to-pipeline